### PR TITLE
6 | Modal menu

### LINKED ISSUE #6: section: hero | menu
Image

Image

### PR DESCRIPTION
**Pull request by ohiienko-r**
PR resolves: https://github.com/BUI-studies/buoy-landing/issues/6

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -12,6 +15,8 @@ function App() {
 			<div>
 				<p>ПОРУЧ ДІТИ</p>
 			</div>
```
Comment (yakovenkoandyUA) [inline src/App.tsx:17]: remove it from the main component, and make a human structure with header main and footer, where main will contain just a set of sections

File: src/components/ModalMenu/ModalMenu.tsx [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+interface ModalMenuProps {
+	setVisible: Dispatch<SetStateAction<boolean>>
+}
+
+const ModalMenu: FC<ModalMenuProps> = ({ setVisible }) => {
+	return (
+		<div className="modal-wrapper">
+			<div className="menu">
+				<div className="menu__close-btn-container">
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/ModalMenu.tsx:12]: without container, just menu__close

File: src/components/ModalMenu/ModalMenu.tsx [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+							strokeLinecap="round"
+							strokeLinejoin="round"
+						/>
+					</svg>
+				</div>
+				<div className="menu__list-container">
+					<ul className="menu__list">
+						<li className="menu__list-item active">
+							<a href="#" className="t__menu-item menu__list-item-link">
+								Головна
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/ModalMenu.tsx:37]: this is 'menu__list-link'

File: src/components/ModalMenu/ModalMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+							</a>
+						</li>
+						<li className="menu__list-item">
+							<a href="#" className="t__menu-item menu__list-item-link">
+								Контакти
+							</a>
+						</li>
+					</ul>
+				</div>
+				<div className="menu__social-links-container">
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/ModalMenu.tsx:57]: without container, just menu__socials

File: src/components/ModalMenu/ModalMenu.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
+import { Dispatch, FC, SetStateAction } from 'react'
+import './styles.scss'
+
+interface ModalMenuProps {
+	setVisible: Dispatch<SetStateAction<boolean>>
+}
+
+const ModalMenu: FC<ModalMenuProps> = ({ setVisible }) => {
+	return (
+		<div className="modal-wrapper">
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/ModalMenu.tsx:10]: modal haven't transition to show

File: src/components/ModalMenu/styles.scss [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+	width: 100vw;
+	height: 100vh;
+	@include flex();
+	background-color: rgba(0, 0, 0, 0.35);
+	backdrop-filter: blur(3px);
+}
+
+.menu {
+	max-width: 1105px;
+	max-height: 735px;
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/styles.scss:23]: we do not set the height of the elements

File: src/components/ModalMenu/styles.scss [hunk 7]
```diff
@@ -0,0 +1,114 @@
...
+	width: 100%;
+	height: 100%;
+	padding: 3.5em;
+	display: flex;
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/styles.scss:27]: why do you use flex if the elements are already in a column

File: src/components/ModalMenu/styles.scss [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+			path {
+				stroke: $primary-color;
+			}
+		}
+	}
+	&__list-container {
+		@include flex();
+	}
+	&__list {
+		display: flex;
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/styles.scss:72]: again element are already in a column, just use margin

File: src/components/ModalMenu/styles.scss [hunk 9]
```diff
@@ -0,0 +1,114 @@
...
+		flex-direction: column;
+		gap: 2em;
+		@media screen and (max-width: 576px) {
+			gap: 1em;
+		}
+
+		&-item-link {
+			color: $typography-color;
+			display: inline-block;
```
Comment (yakovenkoandyUA) [inline src/components/ModalMenu/styles.scss:81]: inline block for what?

**Review by yakovenkoandyUA — CHANGES_REQUESTED** [review]

File: src/App.tsx (hunk 1 above)
Comment (ohiienko-r) [reply src/App.tsx:17]: This PR has to be reviewed after previous one due to all changes to structure and so on are made there

**Review by yakovenkoandyUA — APPROVED** [review]

**Pull request merged**